crates/core/c8y_api/src/smartrest/smartrest_serializer.rs

```diff
@@ -411,21 +413,21 @@ mod tests {
     fn serialize_smartrest_set_operation_to_executing() {
         let smartrest =
             set_operation_executing_with_name(CumulocitySupportedOperations::C8ySoftwareUpdate);
-        assert_eq!(smartrest, "501,c8y_SoftwareUpdate");
+        assert_eq!(smartrest.as_str(), "501,c8y_SoftwareUpdate");
```

Comment:
Would it have been a good idea to provide a `Deref<Target=str>` impl for the `SmartrestPayload` type, to ease these kinds of comparisons, without having to call `as_str()` everywhere?

Reply:
Tried it and it didn't work, even with `Deref<Type = str>`, still needed to call `.as_str()` to compare as string slices.
I think this is because, from the docs:

> If T implements Deref<Target = U>, and v is a value of type T, then:
> - ...
> - T implicitly implements all the methods of the type U which take the &self receiver.

For comparisons, `==` is sugar for `PartialEq::eq`, we'd have to use `str::eq`, but because we implement `PartialEq`, `SmartrestPayload::eq` is used instead.